packages/core/core-flows/src/cart/workflows/update-line-item-in-cart.ts

```diff
@@ -73,6 +79,30 @@ export const updateLineItemInCartWorkflow = createWorkflow(
       return [item.variant_id].filter(Boolean)
     })
 
+    const setPricingContext = createHook(
+      "setPricingContext",
+      {
+        cart,
+        item,
+        variantIds,
+        additional_data: input.additional_data,
+      },
+      {
+        resultValidator: pricingContextResult,
+      }
+    )
+
+    const setPricingContextResult = setPricingContext.getResult()
+    const pricingContext = transform(
+      { cart, setPricingContextResult },
+      (data) => {
+        return {
+          ...data.cart,
+          ...(data.setPricingContextResult ? data.setPricingContextResult : {}),
```

Comment:
suggestion: I wonder if we should enforce currency, region, and customer here? Now that we are opening up for extensibility, users could mistakenly overwrite those. We probably don't want that to happen.

**Same thought for all cart workflows.**

Reply:
Yup, fixed all those with tests. https://github.com/medusajs/medusa/pull/11891/commits/279d2fa5e86bc37aea796f8118b5f4a8639132ad